manifests/01-validating-admission-policy.yaml

```diff
@@ -20,7 +20,7 @@ spec:
     # Consider only request to Gateway API CRDs.
     - name: "check-only-gateway-api-crds"
       # When the operation is DELETE, the "object" variable is null.
-      expression: "(request.operation == 'DELETE' ? oldObject : object).spec.group == 'gateway.networking.k8s.io'"
+      expression: "(request.operation == 'DELETE' ? oldObject : object).spec.group == 'gateway.networking.k8s.io' || (request.operation == 'DELETE' ? oldObject : object).spec.group == 'gateway.networking.x-k8s.io'"
```

Comment:
I think you would be better off looking at the CRD channel rather than the group.  Everything in https://github.com/kubernetes-sigs/gateway-api/blob/main/config/crd/experimental/ is experimental, but only 2 CRDs there are using the special group.   See https://github.com/kubernetes-sigs/gateway-api/blob/main/config/crd/experimental/gateway.networking.k8s.io_backendlbpolicies.yaml#L7 for an example of the channel annotation.

Reply:
The reason this PR adds `gateway.networking.x-k8s.io` to the VAP is so that it doesn't miss CRDs that belong to that group.  That is, we're not trying to discriminate between standard and experimental CRDs, and the goal isn't to check specifically for experimental Gateway API CRDs; rather, the goal is to make sure that when the VAP checks Gateway API CRDs, we include standard _as well as_ experimental CRDs when checking Gateway API CRDs.  

Hypothetically, if Gateway API defined yet another API group for Gateway API CRDs, we would need to add that group here as well.  For the purposes of this VAP, we don't care about the channel (standard, experimental or some yet-to-be-defined channel); we just want to make sure the VAP covers _all_ Gateway API CRDs.